deps/rabbit/src/rabbit_classic_queue_index_v2.erl

```diff
@@ -1277,8 +1278,8 @@ queue_name_to_dir_name(#resource { kind = queue,
     rabbit_misc:format("~.36B", [Num]).
 
 segment_file(Segment, #qi{ dir = Dir }) ->
-    filename:join(rabbit_file:binary_to_filename(Dir),
-                  integer_to_list(Segment) ++ ?SEGMENT_EXTENSION).
+    N = integer_to_binary(Segment),
```

Comment:
There is no need to convert Dir to a list, `filename:join/2` can take a binary as an arg and as soon as one of the args is a binary the output will also be binary.

Reply:
The original code likely was a copy paste from older rabbit code where it used to be necessary.